src/home/NotificationCenter.test.tsx

```diff
@@ -0,0 +1,498 @@
+  },
+  home: {
+    notifications: {
+      testNotification: {
+        ...testNotification,
+        dismissed: true,
+      },
+    },
+  },
+}
+
+const testReward = {
+  amount: '2',
+  contractAddress: 'contractAddress',
+  createdAt: Date.now(),
+  index: 0,
+  proof: [],
+  tokenAddress: '0xcusd',
+}
+
+const superchargeSetUp = {
+  web3: {
+    account: 'account',
+  },
+  app: {
+    numberVerified: true,
+  },
+  supercharge: {
+    availableRewards: [testReward],
+  },
+}
+
+const superchargeWithoutRewardsSetUp = {
+  ...superchargeSetUp,
+  supercharge: {
+    availableRewards: [],
+  },
+}
+
+const mockcUsdBalance = {
+  [mockCusdAddress]: {
+    address: mockCusdAddress,
+    isCoreToken: true,
+    balance: '100',
+    symbol: 'cUSD',
+    usdPrice: '1',
+    priceFetchedAt: Date.now(),
+  },
+}
+
+const mockcUsdWithoutEnoughBalance = {
+  [mockCusdAddress]: {
+    address: mockCusdAddress,
+    isCoreToken: true,
+    balance: '5',
+    symbol: 'cUSD',
+    usdPrice: '1',
+    priceFetchedAt: Date.now(),
+  },
+}
+
+describe('NotificationCenter', () => {
+  it('renders empty state when there is no notifications at all', () => {
+    const store = createMockStore({ ...storeDataNotificationsDisabled })
+    const { queryByTestId, queryByText } = render(
+      <Provider store={store}>
+        <NotificationCenter {...getMockStackScreenProps(Screens.NotificationCenter)} />
+      </Provider>
+    )
+
+    expect(queryByTestId('NotificationCenter/EmptyState')).toBeTruthy()
+    expect(queryByText('youAreAllCaughtUp')).toBeTruthy()
+  })
+
+  it('renders reverify notification if decentrally verified and not CPV', () => {
+    const store = createMockStore({
+      app: {
+        requireCPV: true,
+        numberVerified: true,
+        phoneNumberVerified: false,
+      },
+    })
+    const { getByText } = render(
+      <Provider store={store}>
+        <NotificationCenter {...getMockStackScreenProps(Screens.NotificationCenter)} />
+      </Provider>
+    )
+
+    expect(getByText('reverifyUsingCPVHomecard.description')).toBeTruthy()
+
+    fireEvent.press(getByText('reverifyUsingCPVHomecard.buttonLabel'))
+    expect(navigate).toHaveBeenCalledWith(Screens.VerificationStartScreen, {
+      hideOnboardingStep: true,
+    })
+  })
+
+  it('renders incoming payment request when they exist', () => {
+    const store = createMockStore({
+      ...storeDataNotificationsDisabled,
+      account: {
+        ...storeDataNotificationsDisabled.account,
+      },
+      paymentRequest: {
+        incomingPaymentRequests: [mockPaymentRequests[0]],
+      },
+    })
+    const { getByTestId } = render(
+      <Provider store={store}>
+        <NotificationCenter {...getMockStackScreenProps(Screens.NotificationCenter)} />
+      </Provider>
+    )
+
+    const titleElement = getByTestId('IncomingPaymentRequestNotification/FAKE_ID_1/Title')
+    expect(getElementText(titleElement)).toBe('incomingPaymentRequestNotificationTitle, {}')
```

Comment:
is `incomingPaymentRequestNotificationTitle` meant to have an empty argument here?

Reply:
good catch! I've updated the mock state to have an address->name mapping here and in the sibling test for NotificationBox